api/v1/common_types.go

```diff
@@ -33,13 +42,18 @@ type BuildGit struct {
 }
 
 type BuildStatus struct {
-	// the Signed upload URL
+	// UploadURL is a signed URL to upload the tar'd docker build context to.
 	UploadURL string `json:"uploadURL,omitempty"`
-	// Md5Checksum is the last md5 checksum that resulted in the successful creation of an UploadURL.
-	// +kubebuilder:validation:MaxLength=32
-	// +kubebuilder:validation:MinLength=32
-	// +kubebuilder:validation:Pattern="^[a-fA-F0-9]{32}$"
-	Md5Checksum string `json:"md5checksum,omitempty"`
+
+	// UploadExpiration is the time at which the UploadURL expires.
+	UploadExpiration metav1.Time `json:"uploadExpiration,omitempty"`
```

Comment:
I didn't think we needed this since it can be extracted from the signed URL itself. Maybe I'll find the reason deeper in the review...

Reply:
Finished the review and I'm still not totally sure we need the expiration but open to how you were reasoning about it. I see why the logic to extract the expiration was removed (provider specifics belong in SCI implementations) but still not clear why a client needs the expiration